src/EditorFeatures/TestUtilities/Workspaces/TestWorkspace.cs

```diff
@@ -354,6 +354,18 @@ protected override void ApplyAnalyzerConfigDocumentRemoved(DocumentId documentId
             this.OnAnalyzerConfigDocumentRemoved(documentId);
         }
 
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
+        {
+            if (projectChanges.OldProject.FilePath != projectChanges.NewProject.FilePath)
+            {
+                var hostProject = this.GetTestProject(projectChanges.NewProject.Id);
+                hostProject.OnProjectFilePathChanged(projectChanges.NewProject.FilePath);
+                base.OnProjectNameChanged(projectChanges.NewProject.Id, projectChanges.NewProject.Name, projectChanges.NewProject.FilePath);
+            }
+
+            base.ApplyProjectChanges(projectChanges);
```

Comment:
why was this changed?

Reply:
`base.ApplyProjectChanges` does not handle project name changes, and it does not lead to updating the project name on the test project. This functionality is needed for CodeStyle layer tests based on this test framework.